src/fns.ts

```diff
@@ -46,12 +50,16 @@ export const rateLimit = <T = unknown>(limit: number) => {
     // See: https://runkit.com/dubiousdavid/handling-promise-rejections
     prom.then(
       () => {
-        debugRL('delete')
-        // Remove from the set after resolving
+        debugRL('resolved')
+        // Remove from the set
         set.delete(prom)
       },
       // Handle the exception so we don't throw an UnhandledPromiseRejection exception
-      () => debugRL('exception thrown')
+      () => {
+        debugRL('rejected')
+        // Remove from the set
+        set.delete(prom)
```

Comment:
It would be even better if we exposed the actual errors, instead of just the count. Then the caller could e.g. log them.

Reply:
> It would be even better if we exposed the actual errors, instead of just the count. Then the caller could e.g. log them.

Yes, I also had the same thought since the rate limit util is pretty opinionated in that any promises you give it have to handle their rejections. In most cases, I don't think you would want to log the unhandled exceptions but maybe even just throw whatever errors we got so that the execution after the rate limit usage doesn't continue and assume every promise completed successfully.

Reply:
This code has swallowed errors for a while. Allowing them to fill up the `Set` was a bug and not a feature. I think we might be overthinking things here. We handle errors per promise for all uses of `rateLimit` in our codebase that I have seen. What's the use case for capturing the errors and how is that beneficial?

Reply:
That's a good point. I think the way it is, the caller already has everything they need in order to be able to handle errors in their own code, so we don't really need to build anything extra here to facilitate error handling.

Reply:
Sounds good, as long as the user is aware that the rateLimit acts like a `Promise.allSettled` and they need to do any error handling we don't have to add anything